### PR TITLE
Reader: combined cards must have been posted the same day relative to…

### PR DESCRIPTION
**Pull request by samouri**
… the user or else they won't getcombined

Implements: https://github.com/Automattic/wp-calypso/issues/12151

The strategy here was to attach a moment to each postKey as it is ingested in the keyMakers.  the flag `local()` says to use the users local time.

### CODE REVIEW

File: client/reader/stream/index.jsx [hunk 1]
```diff
@@ -92,6 +92,10 @@ function sameSite( postKey1, postKey2 ) {
 		);
 }
 
+function sameDay( postKey1, postKey2 ) {
+	return postKey1.localMoment.date() === postKey2.localMoment.date(); // date === day of the month
```
Comment (blowery) [inline client/reader/stream/index.jsx]: might want to include the month here too. given how we use it, collisions are unlikely, but better safe than weird.

File: client/lib/feed-stream-store/index.js [hunk 2]
```diff
@@ -19,22 +20,25 @@ const wpcomUndoc = wpcom.undocumented();
 function feedKeyMaker( post ) {
 	return {
 		feedId: post.feed_ID,
-		postId: post.ID
+		postId: post.ID,
+		localMoment: moment( post.date ).local(),
```
Comment (blowery) [inline client/lib/feed-stream-store/index.js]: looks like you don't really need the `.local()` calls https://momentjs.com/docs/#/parsing/utc/

We'd have to use `.utc()` explicitly to get into utc mode.

developer_tools_-_https___momentjs_com_docs_

File: client/lib/feed-stream-store/index.js [hunk 3]
```diff
@@ -21,15 +21,15 @@ function feedKeyMaker( post ) {
...
-		localMoment: moment( post.date ).local(),
+		localMoment: moment( post.date ),
 	};
 }
 
 function siteKeyMaker( post ) {
 	return {
 		blogId: post.site_ID,
 		postId: post.ID,
-		localMoment: moment( post.date ).local(),
```
Comment (samouri) [inline client/lib/feed-stream-store/index.js]: why did you remove `local` calls? is that the default?

Comment (samouri) [inline client/lib/feed-stream-store/index.js]: > By default, moment parses and displays in local time.

from https://momentjs.com/docs/#/parsing/utc/.
awesome

**Comment by blowery** [comment]
fixed it. typo on the date compare.
